packages/next/bin/next.ts

```diff
@@ -129,7 +129,16 @@ if (process.versions.pnp === '3') {
 }
 
 // Make sure commands gracefully respect termination signals (e.g. from Docker)
-process.on('SIGTERM', () => process.exit(0))
+// Allow the graceful termination to be manually configurable
+if (!process.env.NEXT_MANUAL_SIGTERM) {
+  process.on('SIGTERM', () =>
+    setTimeout(
+      () => process.exit(0),
+      parseInt(process.env.NEXT_GRACEFUL_SHUTDOWN_TIMEOUT_MS || '0', 10)
```

Comment:
This timeout shouldn't be added, if `NEXT_MANUAL_SIGTERM` is set, the user should add their own `SIGTERM` handling removing the need for an arbitrary timeout like mentioned in the requested changes here https://github.com/vercel/next.js/pull/29959#discussion_r803265324

Reply:
Makes sense I will make this change, I left it so even if the user doesn't add a manual graceful shutdown handler in the `_document ` but still wants to add an arbitiary delay.

 Now that I think about it sounds kinda like a stupid inconsistency. 😆

Reply:
I see, let's also include the SIGINT in the env variable check, might make sense to call it something like NEXT_MANUAL_SIG_HANDLE

And we can also mention this env variable in the deployment doc here https://github.com/vercel/next.js/blob/canary/docs/deployment.md#nodejs-server

Reply:
```js
if (!process.env.NEXT_MANUAL_SIG_HANDLE) {
  process.on('SIGTERM', () => process.exit(0))
  process.on('SIGINT', () => process.exit(0))
}
```
Like this?

Reply:
Yeah exactly

Reply:
Cool, will update the docs now

Reply:
Done